osidb/models.py

```diff
@@ -1753,6 +1764,25 @@ def create_flawcomment(flaw, external_system_id, comment, **extra_fields):
             flawcomment.meta_attr = comment
             return flawcomment
         except ObjectDoesNotExist:
+            # The "text" field is required for full validation, so save it if the caller of
+            # this method forgets to explicitly copy it over from meta_attr.
+            extra_fields["text"] = extra_fields.get("text") or comment.get("text")
+
+            # The "order" field is required for full validation, but bugzilla calls it "count".
+            extra_fields["order"] = extra_fields.get("order") or comment.get("count")
+
+            # In-place upgrade of a temporary pending comment to preserve uuid.
```

Comment:
I don't think we'll be able to do this if we don't require order, but that's ok, we can just delete all pending comments and recreate them with the data provided by bzimport.

Reply:
My goal was to make the POST endpoint return the actual uuid of the new comment. But yeah, if concurrent requests are made, multiple pending comments are going to be created. Perhaps a simple way to resolve that would be to just delete all pending comments in FlawSaver cleanup. Pending comments are marked by blank `external_system_id` and it wasn't blank-able until now, so I think bugzilla will never provide a blank value in there.

Reply:
Cleanup added in commit a3415f9b5cde43c7e4fb3ac8476e8791fd710365 - https://github.com/RedHatProductSecurity/osidb/pull/209/commits/a3415f9b5cde43c7e4fb3ac8476e8791fd710365#diff-ebe1c224bdd0ffc9f247b35caecd1b03746276b968e7cbf57200af9c5958cd76R253

Reply:
Actually, in commit 3e83eb9 I removed the cleanup and added locking with a comment that explains it. Delete & recreate without that locking wouldn't help because multiple concurrent transactions would simply delete & recreate in a way that would be correct from each transaction's perspective, but invalid in composition after transactions are committed. The reason is that transactions in django don't lock the table, again detailed in the comment in commit 3e83eb9 and in comment https://github.com/RedHatProductSecurity/osidb/pull/209#issuecomment-1574889681